realm/realm-library/src/androidTest/java/io/realm/RealmObjectSchemaTests.java

```diff
@@ -730,6 +731,28 @@ public void getPrimaryKey_nonExistFieldThrows() {
         schema.getPrimaryKey();
     }
 
+    @Test
+    public void getFieldIndex() {
```

Comment:
This unit test targets an internal method. It would be more future proof if the unit test tested the expected behavior from the public API instead.

E.g, I think this PR will instead of "Class `Thing` does not exists in the Schema" say "Field `NoField` does not exists"?